WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java

```diff
@@ -575,25 +574,28 @@ public void fillFeedingForm(String weightVal, Integer index)
 
     public void fillWeightForm(String weightVal, Integer index)
     {
-        fillAnInput("animalid_" + index.toString(), ANIMAL_SUBSET_EHR_TEST[index]);
+        String animalId = ANIMAL_SUBSET_EHR_TEST[index];
+        fillAnInput("animalid_" + index, animalId );
         WebElement el2 = fillAnInput("weight_" + index.toString(), weightVal);
 
         //commenting out since this tries to tab over the date and time and fails, since looks like it requires selecting the date and time value
 //        el2.sendKeys(Keys.TAB);
 //        el2.sendKeys(Keys.TAB);
 
-        fillAnInput("remark_"+ index.toString(), "Entered from automated test");
-        fillAnInput("restraint_" + index.toString(), "T");
+        fillAnInput("remark_"+ index, "Entered from automated test");
+        fillAnInput("restraint_" + index, "T");
+        waitForElement(Locator.linkWithText(animalId), WAIT_FOR_PAGE * 2); // Wait for the Animal Info window to load
     }
 
     public void waitUntilElementIsClickable(String id)
     {
         shortWait().until(ExpectedConditions.elementToBeClickable(Locator.id(id)));
     }
 
-    public void clickNewButton(String id){
-        WebElement o = Locator.tagWithId("button",id).findElement(getDriver());
-        o.click();
+    public void clickNewButton(String id)
+    {
+        waitUntilElementIsClickable(id);
+        Locator.tagWithId("button", id).findElement(getDriver()).click();
```

Comment:
Make `waitUntilElementIsClickable` return the element it waits for. Then you can just click what it returns.
```suggestion
    public void clickNewButton(String id)
    {
        waitUntilElementIsClickable(id).click();
```